src/firefly/java/edu/caltech/ipac/visualize/plot/ImagePlot.java

```diff
@@ -242,11 +232,12 @@ public void preProcessImageTiles(final ActiveFitsReadGroup frGroup) {
         if (_imageData.isUpToDate()) return;
         synchronized (this) {
             if (_imageData.isUpToDate()) return;
-            if (_imageData.size()<4 || CORE_CNT==1) {
+            int coreCnt= ServerContext.getParallelProcessingCoreCnt();
+            if (_imageData.size()<4 || coreCnt==1) {
                 for(ImageData id : _imageData)  id.getImage(frGroup.getFitsReadAry());
             }
             else {
-                ExecutorService executor = Executors.newFixedThreadPool(CORE_CNT);
+                ExecutorService executor = Executors.newFixedThreadPool(coreCnt);
```

Comment:
`executor` should be static.  `coreCnt` is shared among threads, and should not be assumed single threaded.